tests/clar_libgit2.c

```diff
@@ -158,6 +158,18 @@ int cl_rename(const char *source, const char *dest)
 
 #endif
 
+int cl_setenv_printf(const char *name, const char *fmt, ...)
+{
+	va_list args;
+	git_buf buf = GIT_BUF_INIT;
+
+	va_start(args, fmt);
+	cl_git_pass(git_buf_vprintf(&buf, fmt, args));
+	va_end(args);
+
+	return cl_setenv(name, git_buf_cstr(&buf));
```

Comment:
Good catch. I'd argue that's a good reason to *have* the helper method, because we can fix any bugs in one place and avoid duplicating the same code pattern repeatedly.  The new test suite bits would get significantly longer if they have to duplicate this logic many times over.

I'd really like to fix this bug and keep the method.